HopsanGUI/HcomHandler.cpp

```diff
@@ -6439,154 +6442,126 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
         }
 
+
+
         //Parse windowing function argument
-        WindowingFunctionEnumT windowingFunction;
-        if(!windowingFuncArg.isEmpty()) {
-            if(windowingFuncArg.toLower() == "hann") {
+        WindowingFunctionEnumT windowingFunction = RectangularWindow;
+        if(splitArgs.size() > i) {
+            if(splitArgs[i].toLower() == "hann") {
                 windowingFunction = HannWindow;
+                ++i;
             }
-            else if(windowingFuncArg.toLower() == "rectangular") {
+            else if(splitArgs[i].toLower() == "rectangular") {
                 windowingFunction = RectangularWindow;
+                ++i;
             }
-            else if(windowingFuncArg.toLower() == "flattop") {
+            else if(splitArgs[i].toLower() == "flattop") {
                 windowingFunction = FlatTopWindow;
-            }
-            else {
-                HCOMERR("Unknown windowing function: "+windowingFuncArg);
-                mAnsType = Undefined;
-                return;
+                ++i;
             }
         }
 
-        //Parse window min and max time arguments
-        double minTime, maxTime;
-        if(!minTimeArg.isEmpty()) {
-            if(maxTimeArg.isEmpty()) {
-                HCOMERR("Maximum time for windowing is required when minimum time is specified");
-                mAnsType = Undefined;
-                return;
-            }
+        double minTime=-std::numeric_limits<double>::max();
+        double maxTime=std::numeric_limits<double>::max();
+
+        if(splitArgs.size() > i+1) {
             bool ok;
-            minTime = getNumber(minTimeArg, &ok);
+            minTime = getNumber(splitArgs[i], &ok);
             if(!ok) {
-                HCOMERR("Unknown time limit: "+minTimeArg);
+                HCOMERR("Unknown minimum time limit: "+splitArgs[i]);
                 mAnsType = Undefined;
                 return;
             }
-            maxTime = getNumber(maxTimeArg, &ok);
+            maxTime = getNumber(splitArgs[i+1], &ok);
             if(!ok) {
-                HCOMERR("Unknown time limit: "+maxTimeArg);
+                HCOMERR("Unknown maximum time limit: "+splitArgs[i+1]);
                 mAnsType = Undefined;
                 return;
             }
+            i+=2;
         }
 
-        if(typeArg.isEmpty()) {
-            mAnsVector = pDataVar->toFrequencySpectrum(pTimeVar, PowerSpectrum);
+        if(splitArgs.size() > i) {
+            HCOMERR("Unknown argument: "+splitArgs[i]);
+            mAnsType = Undefined;
+            return;
         }
-        else if(windowingFuncArg.isEmpty()) {
-            mAnsVector = pDataVar->toFrequencySpectrum(pTimeVar, type);
+
+        mAnsType = DataVector;
+        mAnsVector = pVar->toFrequencySpectrum(pTimeVar, type, windowingFunction, minTime, maxTime);
+        return;
+    }
+    else if(isHcomFunctionCall("rms", expr))
+    {
+        QString args = expr.mid(4, expr.size()-5);
```

Comment:
Can we use extractFunctionCallExpressionArguments( instead of expr.mid ?
or does it not work with nested parentheses ?

Reply:
Good point. It does handle nested parentheses internally, so that is no problem.